lib/source/defender/aws_iot_defender_mqtt.c

```diff
@@ -113,19 +115,31 @@ void AwsIotDefenderInternal_DeleteTopicsNames( void )
 
 /*-----------------------------------------------------------*/
 
-IotMqttError_t AwsIotDefenderInternal_MqttConnect( void )
+void AwsIotDefenderInternal_SetMqttConnection( IotMqttConnection_t mqttConnection )
```

Comment:
This internal function doesn't seem to have much use.  It's setting an internal library variable, but you could just declare the variable in the library's internal header.